runtime/parachains/src/disputes.rs

```diff
@@ -601,34 +666,55 @@ impl<T: Config> From<VoteImportError> for Error<T> {
 struct ImportUndo {
 	/// The validator index to which to associate the statement import.
 	validator_index: ValidatorIndex,
-	/// The direction of the vote, for block validity (`true`) or invalidity (`false`).
-	valid: bool,
+	/// The kind and direction of the vote.
+	vote_kind: VoteKind,
 	/// Has the validator participated before, i.e. in backing or
 	/// with an opposing vote.
 	new_participant: bool,
 }
 
 struct DisputeStateImporter<BlockNumber> {
 	state: DisputeState<BlockNumber>,
+	backers: BTreeSet<ValidatorIndex>,
 	now: BlockNumber,
 	new_participants: bitvec::vec::BitVec<u8, BitOrderLsb0>,
 	pre_flags: DisputeStateFlags,
+	pre_state: DisputeState<BlockNumber>,
```

Comment:
Some doc comments would be nice.

Reply:
Add docs for `pre_backing` in [b8de8d6](https://github.com/paritytech/polkadot/pull/6249/commits/b8de8d6691d4e3437321b0c790ba619587762ecb). Let me know if you want some docs for `pre_state`.